### ISSUE TITLE
Change to keyboard navigation order

### ISSUE THREAD

**Issue by jessegreenberg** [phetsims/reactants-products-and-leftovers#56]
@zepumph after a discussion with @emily-phet we would like to change the tab order, would you mind changing to this order?

From the design doc, here is the requested order:

Sandwiches Screen:
 - Radio Buttons
 - “Custom” number spinners (when visible)
 - Before AccordionBox
 - After AccordionBox
 - reset all button

Molecules Screen:
 - Radio Buttons
 - Before AccordionBox
 - After AccordionBox
 - reset all button

Game Screens:
 - NumberSpinners
 - “Check/try again/show answer” buttons
 - “Start Over” button

**Comment by zepumph** [comment]
I updated these as expressed above, here is the latest link:

**Comment by terracoda** [comment]
@zepumph, regarding comment 1 above, my instinct would be that, greyed out, but still visible buttons, should remain findable (possibly remain in focus order), but be communicated clearly as disabled. Whereas, buttons that completely disappear visually would make more sense to remove from the focus order. 

Regarding comment 2, my instinct would be that focus has to stay on the button until the user moves their focus. However, again, I will have to play more with the sim in the morning.

**Comment by terracoda** [comment]
@zepumph, in a general sense the Tab Order is fine. Groupings, however, will be useful. I'm not sure the best way to markup a group of radio buttons (still need to work this out); however, the accepted radio group interaction pattern is as follows:
- Tab to the group, placing focus on the first item (if none are selected) or the selected item if one is selected
- Use Up and Left Arrow keys to move to previous radio
- Use Down and Left Arrow keys to move to the next radio
- Each radio is auto-selected upon focus

Examples in issue https://github.com/phetsims/a11y-research/issues/57

And for the spinner buttons, I think it makes sense to place focus around the entire control  (text box, and arrow buttons) if the text input will allow direct key input, and then up and down arrows would then increase and decrease the value in the box.  Will it be possible to enter a number? That would be ideal. If not, I think the focus could just go around the arrow buttons, so it is clear the users have to use the arrow keys (Up and Down) to increase and decrease the value in the box. The `spinbutton` is a native HTML `input` with the `type="number"`.

There are other native hotkeys that come with it as there are with sliders. It might make sense to make a spinbutton that fits PhET.
 
The ARIA spec describes the [spinbutton pattern](https://www.w3.org/TR/wai-aria-practices-1.1/#spinbutton).

**Comment by pixelzoom** [comment]
@terracoda asked:
> And for the spinner buttons, ... Will it be possible to enter a number? That would be ideal.  ...

Entry of numbers and text is generally not supported throughout PhET sims.

**Comment by terracoda** [comment]
@pixelzoom, ok, good to know, but as we add in accessibility features, we might want to rethink that interaction. Direct input can be quite useful in some cases.

**Comment by pixelzoom** [comment]
To elaborate...  If I recall correctly, the decision not to include keyboard input in PhET sims was motivated by (1) the desire to interact on mobile devices without keyboard, due in part to the lousy user experience that keyboards provide on mobile device, and (2) engineering challenges and costs with supporting keyboard input.   So PhET's library of user interface components does not generally support keyboard input.   We can certainly revisit that decision, but adding keyboard support will be a costly engineering effort.

**Comment by zepumph** [comment]
Thoughts on RPAL from keyboard nav meeting:

* Get the radio button keyboard nav in order. Currently we have `<li>` tags around inputs without a `<ul>`
* Get `NumberSpinner` keyboard nav to act like `NumberPicker` where arrow keys can adjust the number values (see gravity-force-lab)

**Comment by pixelzoom** [comment]
@zepumph said:
> Get `NumberSpinner` keyboard nav to act like `NumberPicker` where arrow keys can adjust the number values (see gravity-force-lab)

Thoughts on spinners and pickers...  `NumberSpinner` was implemented first, patterned on Java spinners, with arrows on the sides.  Then we decided we needed something different, with arrows on the top and bottom. Someone coined the term "picker", which I really don’t like, but it unfortunately stuck, and thus was born `NumberPicker`.  Their behavior started to diverge from there - for example `NumberPicker` allows you to click on the top/bottom of the value, while `NumberSpinner` does not.  In terms of standard UI terminology, `NumberSpinner ` and `NumberPicker` are both "spinners", and they should behave the same.  Their implementations should someday be combined into one thing called "spinner", with an option for where arrows are placed.

**Comment by jessegreenberg** [comment]
Ah, thanks for the history @pixelzoom. Regarding

> In terms of standard UI terminology, NumberSpinner and NumberPicker are both "spinners", and they should behave the same

We came to the same conclusion, and since they perform the  same UI purpose we determined they should  behave the same in keyboard nav.

@zepumph can this be closed? We will look into the issues with these radio buttons in https://github.com/phetsims/sun/issues/316 and navigation for NumberSpinner will be handled in https://github.com/phetsims/sun/issues/318.

**Comment by zepumph** [comment]
Sounds good. Closing
